src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5608,7 +5608,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
     <value>Explicit application of 'System.Runtime.CompilerServices.NullableAttribute' is not allowed.</value>
   </data>
   <data name="ERR_NullableUnconstrainedTypeParameter" xml:space="preserve">
-    <value>A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.</value>
+    <value>A nullable type parameter must be known to be a value type or non-nullable reference type. Please use language version '{0}' or greater to use annotations with unconstrained type parameters, or consider adding a 'class', 'struct', or type constraint.</value>
```

Comment:
>unconstrained [](start = 171, length = 13)

I just realized that this term is imprecise, even though we use it in conversations about the feature. Perhaps, "not known to be a reference or value type"?

Reply:
Or even "not known to be a not nullable reference or value type"

---
In reply to: [456699740](https://github.com/dotnet/roslyn/pull/45993#discussion_r456699740) [](ancestors = 456699740)